device/iot-device-samples/file-upload-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/FileUploadSimpleSample.java

```diff
@@ -58,29 +62,64 @@ public static void main(String[] args)
         {
-
             File file = new File(fullFileName);
-            if(file.isDirectory())
+            if (file.isDirectory())
             {
                 throw new IllegalArgumentException(fullFileName + " is a directory, please provide a single file name, or use the FileUploadSample to upload directories.");
             }
-            else
+
+            System.out.println("Retrieving SAS URI from IoT Hub...");
+            FileUploadSasUriResponse sasUriResponse = client.getFileUploadSasUri(new FileUploadSasUriRequest(file.getName()));
+
+            System.out.println("Successfully got SAS URI from IoT Hub");
+            System.out.println("Correlation Id: " + sasUriResponse.getCorrelationId());
+            System.out.println("Container name: " + sasUriResponse.getContainerName());
+            System.out.println("Blob name: " + sasUriResponse.getBlobName());
+            System.out.println("Blob Uri: " + sasUriResponse.getBlobUri());
+
+            System.out.println("Using the Azure Storage SDK to upload file to Azure Storage...");
+
+            try (FileInputStream fileInputStream = new FileInputStream(file))
             {
-                client.uploadToBlobAsync(file.getName(), new FileInputStream(file), file.length(), new FileUploadStatusCallBack(), null);
+                // Note that other versions of the Azure Storage SDK can be used here instead. The latest can be found here:
+                // https://github.com/Azure/azure-sdk-for-java/tree/master/sdk/storage#azure-storage-sdk-client-library-for-java
+                CloudBlockBlob blob = new CloudBlockBlob(sasUriResponse.getBlobUri());
+                blob.upload(fileInputStream, file.length());
             }
+            catch (Exception e)
+            {
+                System.out.println("Exception encountered while uploading file to blob: " + e.getMessage());
+
+                System.out.println("Failed to upload file to Azure Storage.");
+
+                System.out.println("Notifying IoT Hub that the SAS URI can be freed and that the file upload failed.");
+
+                // Note that this is done even when the file upload fails. IoT Hub has a fixed number of SAS URIs allowed active
+                // at any given time. Once you are done with the file upload, you should free your SAS URI so that other
+                // SAS URIs can be generated. If a SAS URI is not freed through this API, then it will free itself eventually
+                // based on how long SAS URIs are configured to live on your IoT Hub.
+                FileUploadCompletionNotification completionNotification = new FileUploadCompletionNotification(sasUriResponse.getCorrelationId(), false);
+                client.completeFileUploadAsync(completionNotification);
 
-            System.out.println("File upload started with success");
+                System.out.println("Successfully notified IoT Hub that the SAS URI can be freed, and that the file upload was a failure.");
```

Comment:
nit: the "Successfully" word in the console will be misleading,
Maybe:
"Failed to upload the file. IoT Hub was notified that the SAS URI can be freed"